src/librbd/crypto/luks/FormatLUKSRequest.cc

```diff
@@ -0,0 +1,122 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "FormatLUKSRequest.h"
+#include <errno.h>
+
+#include "common/dout.h"
+#include "librbd/Utils.h"
+#include "librbd/api/Io.h"
+#include "librbd/crypto/luks/Utils.h"
+#include "librbd/io/AioCompletion.h"
+
+#define dout_subsys ceph_subsys_rbd
+#undef dout_prefix
+#define dout_prefix *_dout << "librbd::crypto::luks::FormatLUKSRequest: " \
+                           << this << " " << __func__
+
+namespace librbd {
+namespace crypto {
+namespace luks {
+
+using librbd::util::create_context_callback;
+
+template <typename I>
+FormatLUKSRequest<I>::FormatLUKSRequest(
+        I* image_ctx, const char* passphrase, size_t passphrase_size,
+        Context* on_finish) : m_image_ctx(image_ctx), m_on_finish(on_finish),
+                              m_fd(-1), m_cd(nullptr),
+                              m_passphrase_size(passphrase_size) {
+  m_passphrase = new char[passphrase_size];
+  memcpy(m_passphrase, passphrase, passphrase_size);
+}
+
+template <typename I>
+void FormatLUKSRequest<I>::send() {
+  auto [m_fd, path] = create_anonymous_file("LUKSHeader");
+  if (m_fd == -1) {
+    lderr(m_image_ctx->cct) << "error creating anonymous file: " << errno
```

Comment:
Nit: this seems to break the encapsulation of the magic of that method. Perhaps change its signature to `int create_anonymous_file(const char* name, int* fd, std::string* path)`